authz/client/src/main/java/org/keycloak/authorization/client/ServerBackChannelUrlConfiguration.java

```diff
@@ -0,0 +1,212 @@
+/*
+ *  Copyright 2019 Red Hat, Inc. and/or its affiliates
+ *  and other contributors as indicated by the @author tags.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+package org.keycloak.authorization.client;
+
+import org.keycloak.authorization.client.representation.ServerConfiguration;
+import org.keycloak.common.util.KeycloakUriBuilder;
+import org.keycloak.constants.ServiceUrlConstants;
+
+import java.util.List;
+
+class ServerBackChannelUrlConfiguration extends ServerConfiguration {
+
+    private static final String TOKEN_INTROSPECTION_PATH = "/realms/{realm-name}/protocol/openid-connect/token/introspect";
+    private static final String USER_INFO_PATH = "/realms/{realm-name}/protocol/openid-connect/userinfo";
+
+    private static final String REGISTRATION_PATH = "/realms/{realm-name}/clients-registrations/openid-connect";
+    private static final String RESOURCE_REGISTRATION_PATH = "/realms/{realm-name}/authz/protection/resource_set";
+    private static final String PERMISSION_PATH = "/realms/{realm-name}/authz/protection/permission";
+    private static final String POLICY_PATH = "/realms/{realm-name}/authz/protection/uma-policy";
+
+    private final ServerConfiguration delegate;
+
+
+    private String tokenUrl;
+
+    private String tokenIntrospectionUrl;
+
+    private String userInfoUrl;
+
+    private String logoutUrl;
+
+    private String jwksUrl;
+
+    private String registrationUrl;
+
+    private String resourceRegistrationUrl;
+
+    private String permissionUrl;
+
+    private String policyUrl;
+
+
+    ServerBackChannelUrlConfiguration(ServerConfiguration delegate, String authServerUrl, String realm) {
+        this.delegate = delegate;
+        resolveBackChannelUrls(authServerUrl, realm);
+    }
+
+
+    private void resolveBackChannelUrls(String backChannelUrl, String realm) {
+        KeycloakUriBuilder builder = KeycloakUriBuilder.fromUri(backChannelUrl);
```

Comment:
I am a bit unsure here. Are there reasons not to override the discovered urls with hard coded values?